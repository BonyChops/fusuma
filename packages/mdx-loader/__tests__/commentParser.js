'use strict';

const commentParser = require('../src/commentParser');

describe('fusuma-loader: commentParser', () => {
  test('should return key: value case', () => {
    {
      const input = 'key: value';

      expect(commentParser(input)).toMatchSnapshot();
    }

    {
      const input = 'key:value';

      expect(commentParser(input)).toMatchSnapshot();
    }
  });

  test('should return key: foo,bar,baz case', () => {
    {
      const input = 'key: foo,bar,baz';

      expect(commentParser(input)).toMatchSnapshot();
    }

    {
      const input = 'key:foo, bar,     baz';

      expect(commentParser(input)).toMatchSnapshot();
    }
  });

  test('should return key case', () => {
    {
      const input = 'key';

      expect(commentParser(input)).toMatchSnapshot();
    }

    {
      const input = '   key    ';

      expect(commentParser(input)).toMatchSnapshot();
    }
  });

  test('should return aaa', () => {
    const input = `zzzzz
test1
test2`;

    expect(commentParser(input)).toMatchSnapshot();
  });
});
