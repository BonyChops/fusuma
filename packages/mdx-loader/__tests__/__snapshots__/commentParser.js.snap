// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fusuma-loader: commentParser should return aaa 1`] = `
Object {
  "prefix": "zzzzz",
  "valueArr": Array [
    "test1",
    "test2",
  ],
  "valueStr": "test1
test2",
}
`;

exports[`fusuma-loader: commentParser should return key case 1`] = `
Object {
  "prefix": "key",
  "valueArr": Array [],
  "valueStr": "",
}
`;

exports[`fusuma-loader: commentParser should return key case 2`] = `
Object {
  "prefix": "key",
  "valueArr": Array [],
  "valueStr": "",
}
`;

exports[`fusuma-loader: commentParser should return key: foo,bar,baz case 1`] = `
Object {
  "prefix": "key",
  "valueArr": Array [
    "foo",
    "bar",
    "baz",
  ],
  "valueStr": "foo,bar,baz",
}
`;

exports[`fusuma-loader: commentParser should return key: foo,bar,baz case 2`] = `
Object {
  "prefix": "key",
  "valueArr": Array [
    "foo",
    "bar",
    "baz",
  ],
  "valueStr": "foo,bar,baz",
}
`;

exports[`fusuma-loader: commentParser should return key: value case 1`] = `
Object {
  "prefix": "key",
  "valueArr": Array [
    "value",
  ],
  "valueStr": "value",
}
`;

exports[`fusuma-loader: commentParser should return key: value case 2`] = `
Object {
  "prefix": "key",
  "valueArr": Array [
    "value",
  ],
  "valueStr": "value",
}
`;
